### PR TITLE
Added pFor class

### LINKED ISSUE #33: Implement index based parallel loops like in OpenMP
Implement a `Static` and `Dynamic` schedulers to break up the index-based job in chunks and implement a parallel for (`pFor`) object capable of automatically dispatching the work to be done. This differs from the regular operation of the library since more than one job is assigned to each worker without the need to "respawn" a new worker.

As an example:
```python
from jobdispatcher.index_based_loops import Static, pFor

def myfunc(n: int) :
    return n+1

ncores = 2
results = pFor(Static(ncores))(myfunc, 0, 10)

print(results)
```

should run the function `myfunc` on 2 cores assigning the jobs with index [0, 1, 2, 3, 4] to the first one and [5, 6, 7, 8, 9] to the second one. The returned `result` list should be ordered according to the index. For the reported example the following return is expected: `[1, 2, 3, 4, 5, 6, 7, 8, 9, 10]`. 

The `Dynamic` scheduler should instead break down the workload into smaller chunks of a specified size and feed them as blocks to the workers.

### PR DESCRIPTION
**Pull request by ghost**
Added an early version of the `pFor` class with `Static` and `Dynamic` schedulers to mimic the index-based OpenMP behavior. Added unit test for the index-based-loop submodule using 2 cores to allow automatic run using GitHub actions default runner.

TO DO for the future:
- Documentation update with the new functionality (waiting for the first commit in main of documentation branch)
- Avoid explicitly generating chunk lists to avoid waste of memory for large jobs.
- Check whether the `get_cuncks` function can be substituted by an already built-in chunker.

### CODE REVIEW

**Comment by GES-mpalermo** [comment]
@GES-ppravatto riesci ad aprire una Issue e linkarla a questa pull request?

Nella Issue se riesci potresti mettere un esempio minimo di come funziona la feature (e.g. con esempio base di codice di utilizzo). Fammi sapere, se c'è qualche dubbio sentiamoci su Teams :)

**Comment by ghost** [comment]
Linked issue 33

File: jobdispatcher/index_based_loops.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+        for chunk in range(nchunks):
+            chunks.append([chunk * self.__size + n for n in range(self.__size)])
+
+        if extra != 0:
+            chunks.append([self.__size * nchunks + n for n in range(extra)])
+
+        return chunks
+
+
+class pFor:
```
Comment (GES-mpalermo) [inline jobdispatcher/index_based_loops.py]: @GES-ppravatto as per PEP8, the class name should follow the CamelCase scheme. PFor is not really nice, what about ParallelFor? It's more explicit and clear :) 

What do you think?

**Review by GES-mpalermo — CHANGES_REQUESTED** [review]
Vedasi commento su nome classe pFor. Una volta a convergenza per me possiamo procedere al merge :)

File: jobdispatcher/index_based_loops.py (hunk 1 above)
Comment (ghost) [reply jobdispatcher/index_based_loops.py]: Changed `pFor` to `ParallelFor`.

**Pull request merged**